opencog/moses/main/demo-problems.cc

```diff
@@ -292,7 +292,7 @@ void polynomial_problem::run(option_base* ob)
 	                      cscore,
 	                      pms.opt_params, pms.hc_params, pms.ps_params,
 	                      pms.deme_params, pms.filter_params, pms.meta_params,
-	                      pms.moses_params, pms.mmr_pa);
+	                      pms.moses_params, pms.mmr_pa, id::contin_type, it.get_labels());
```

Comment:
We don't need to specify the labels because the data is auto-generated and `labels` should be empty.
And the output type wasn't specified before. I know the default is `id::boolean` in `meta_...` but it was working anyway[at least for the combo case]? perhaps something is wrong or I miss something.

Reply:
The reason for the output type being passed is, that we had a buggy code [here](https://github.com/singnet/asmoses/blob/atomese2combo-converter/opencog/combo/converter/combo_atomese.cc#L62)  int the atomese_converter that converted everything to predicate node regardless of the data_type of the dataset being passed. So for example if `$1` is generated by moses for a contin dataset `atomese_converter` simply converts it to predicate node when it should be to schema_node. That was one of the reason behind the inconsistent results of asmoses. For the boolean dataset the code did not affect the output at all. but it sure did for some contin datasets. hence we decided to pass the output type from the dataset to the code .  As for the labels it can be auto-generated we can remove that.

Reply:
Update labels cannot be removed either because the labels of the auto generated data is not empty like you said. it has default label `i1`.